gearoenix/math/gx-math-aabb.hpp

```diff
@@ -124,39 +124,67 @@ struct Aabb3 final {
     [[nodiscard]] constexpr std::optional<Element> hit(const Ray3<Element>& ray, const Element d_min) const noexcept
     {
         const auto& ro = ray.get_origin();
+        const auto& rd = ray.get_normalized_direction();
         const auto& rrd = ray.get_reversed_normalized_direction();
         
-        if (fabs(rrd.x) < FLT_EPSILON)
```

Comment:
Please use the C++ one, `std::abs`, it has overrides for both `float` and `double`.